app/services/DynamoChannelTests.scala

```diff
@@ -127,19 +194,115 @@ class DynamoChannelTests(stage: String) extends StrictLogging {
     putAllBatched(writeRequests)
   }
 
-  def deleteTests(testNames: List[String], channel: Channel): ZIO[ZEnv, DynamoPutError, Unit] = {
+  /**
+    * With an UpdateItem request we have to provide an expression to specify each attribute to be updated.
+    * We do this by iterating over the attributes in `item` and building an expression
+    */
+  private def buildUpdateTestExpression(item: Map[String, AttributeValue]): String = {
+    val subExprs = item.foldLeft[List[String]](Nil) { case (acc, (key, value)) =>
+      s"$key = :$key" +: acc
+    }
+    s"set ${subExprs.mkString(", ")} remove lockStatus" // Unlock the test at the same time
+  }
+
+  def updateTest[T <: ChannelTest[T] : Encoder](test: T, channel: Channel, email: String): ZIO[ZEnv, DynamoError, Unit] = {
+    val item = jsonToDynamo(test.asJson).m().asScala.toMap -
+      "priority" -    // Do not update priority
+      "lockStatus" -  // Unlock by removing lockStatus
+      "name" -        // key field
+      "channel"       // key field
+
+    val updateExpression = buildUpdateTestExpression(item)
+
+    val attributeValues = item.map { case (key, value) => s":$key" -> value }
+    // Add email, for the conditional update
+    val attributeValuesWithEmail = attributeValues + (":email" -> AttributeValue.builder.s(email).build)
+
+    val updateRequest = UpdateItemRequest
+      .builder
+      .tableName(tableName)
+      .key(buildKey(channel, test.name))
+      .updateExpression(updateExpression)
+      .expressionAttributeValues(attributeValuesWithEmail.asJava)
+      .conditionExpression("lockStatus.email = :email")
+      .build()
+
+    update(updateRequest)
+  }
+
+  // Returns the value of the bottom priority test - which is the highest value, because 0 is top priority
```

Comment:
I've gone with "top" and "bottom" when referring to priority, because saying "high" and "low" gets confusing